src/OrchardCore.Modules/OrchardCore.Media/Startup.cs

```diff
@@ -48,10 +49,13 @@ public override void ConfigureServices(IServiceCollection services)
             {
                 var shellOptions = serviceProvider.GetRequiredService<IOptions<ShellOptions>>();
                 var shellSettings = serviceProvider.GetRequiredService<ShellSettings>();
-                var env = serviceProvider.GetRequiredService<IHostingEnvironment>();
 
-                string mediaPath = GetMediaPath(env, shellOptions.Value, shellSettings);
-                return new MediaFileStore(new FileSystemStore(mediaPath, shellSettings.RequestUrlPrefix, "/media"));
+                var mediaPath = GetMediaPath(shellOptions.Value, shellSettings);
+                var fileStore = new FileSystemStore(mediaPath);
+
+                var mediaUrlBase = "/" + fileStore.Combine(shellSettings.RequestUrlPrefix, "media"); // Kind of hackish to use IFileStore.Combine() for a URL, I know...
+
```

Comment:
It doesn't hurt me here because filestore normalizes paths with `/`, then there is `"media"` and `RequestUrlPrefix` should only use valid url characters.

Personaly i would only use.

    mediaUrlBase = $"/{ shellSettings.RequestUrlPrefix}/ media";

Or maybe

    mediaUrlBase = $"/{ shellSettings.RequestUrlPrefix.Replace('\\', '/').Trim('/')}/ media";

Reply:
The latter would work, but the former makes too many assumptions. Not sure I like it better than the current code thought...

Reply:
Yes, i understand, i've no preference too.